sphinxext/cond.py

```diff
@@ -29,9 +43,88 @@ def run(self):
 
         return []
 
+class CondPlus(SphinxDirective):
+    """
+    Directive to only include text if the given tag(s) are enabled.
+    """
+    has_content = True
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+    option_spec = {}
+
+    def run(self):
+        config = self.state.document.settings.env.config
+        try:
+            result = config._raw_config['tags'].eval_condition(self.arguments[0])
+        except ValueError as err:
+            raise self.severe(u'Error parsing conditional parsing expression: {}'.format(SafeString(str(err))))
+
+        if result:
+            # Same as util.nested_parse_with_titles but try to handle nested
+            # sections which should be raised higher up the doctree.
+            memo: Any = self.state.memo
+            surrounding_title_styles = memo.title_styles
+            surrounding_section_level = memo.section_level
+            memo.title_styles = []
+            memo.section_level = 0
+
+            node = Element()
+            node.document = self.state.document
+            set_source_info(self, node)
+            try:
+               self.state.nested_parse(self.content, self.content_offset, node, match_titles=True)
+
+               title_styles = memo.title_styles
+               if (not surrounding_title_styles or
+                    not title_styles or
+                    title_styles[0] not in surrounding_title_styles or
+                    not self.state.parent):
+                # No nested sections so no special handling needed.
+                   return node.children
+
+               current_depth = 0
+               parent = self.state.parent
+               while parent:
+                  current_depth +=1
+                  parent = parent.parent
+                  # This should stop at the "Title" element
+
+               # Accounts for the "Title" element?
+               current_depth -= 2
```

Comment:
This value may go negative and will lead to unexpected behavior. It seems we could simplify here.

Reply:
I'm inclined to agree, but also am only sort of following how the depth-pop logic works.

If I have something nested under an H1, I would expect to get up to `current_depth=1`. Then we do -2 (making it -1), subtract another 1 later (making it -2) and add 1 (making it -1). So yeah that seems...wrong.

Will test a little more and see what happens. I am guessing we can simply state that if `current_depth` is <2, do nothing?

Or if it works _in practice_ there might be something about how Sphinx/Docutils treats the node tree where this value always ends up positive (somehow).

Reply:
After some testing I can't seem to trigger the case where this happens.

I've added a simple print statement if the depth ever drops to be negative - that way if it *does* happen we can identify the circumstances in which it happens and correct for it. As it stands I don't know enough about why the `-=2` logic is included to remove or correct for a negative value.